### DISCUSSION TITLE
Is there a native function to list the products in the cart in the same order they were added? I'm referring to the cart provided by the Pimcore E-commerce Framework.

### DISCUSSION THREAD

**Discussion by AlephLau** [pimcore/pimcore#15685] [Q&A / Support] [answered]
Is there no method to sort the items in the shopping cart in the order they were added? In other words, if I keep adding products to my e-commerce, I want them to be listed in my cart in the same order they were added. Is there no native solution for this? Because I think I'll have to add a DateTime field to list them in the order of the date they were added. Please let me know if you have any less intrusive solution on my objects; I believe there should be a native method for something as fundamental as this. Thank you.

**Comment by AlephLau** [comment]
Hello @mattamon , how are you? When you have the chance, do you think you can help me solve this question?

**Comment by jdreesen** [comment, marked as answer]
As far as I know, the items in the cart *are* sorted in the order they were added, see: https://github.com/pimcore/ecommerce-framework-bundle/blob/1.x/src/CartManager/Cart.php#L124

Apart from this, there is [`CartInterface::sortItems()`](https://github.com/pimcore/ecommerce-framework-bundle/blob/1.x/src/CartManager/Cart.php#L67) if you want to sort them in a custom order.

**Reply by KitsuneQki** [reply]
I'm having an issue with the native sort, because all the cart products have the same addedDateTimestamp, how can this be solved?

**Reply by jdreesen** [reply]
> there is [CartInterface::sortItems()](https://github.com/pimcore/ecommerce-framework-bundle/blob/1.x/src/CartManager/Cart.php#L67) if you want to sort them in a custom order.

Did you try this?

**Reply by KitsuneQki** [reply]
I've been trying to fix the addedDateTimestamp first but I can't find where the addedDateTimestamp is set by default and why is adding the same value in all the products, I'm also looking to sort cart items in the order they were added.
